sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolvePartitionSpec.scala

```diff
@@ -51,14 +61,16 @@ object ResolvePartitionSpec extends Rule[LogicalPlan] {
   private def resolvePartitionSpecs(
       tableName: String,
       partSpecs: Seq[PartitionSpec],
-      partSchema: StructType): Seq[ResolvedPartitionSpec] =
+      partSchema: StructType,
+      checkSpec: TablePartitionSpec => Unit = _ => ()): Seq[ResolvedPartitionSpec] =
```

Comment:
Why do we make it so general as a callback function? Shall we just call `requireExactMatchedPartitionSpec` inside this method?

Reply:
We shouldn't call it in all cases, for instance, we shouldn't call it for `SHOW PARTITIONS`.

I thought about alternative solution by passing a flag and check the flag but just passing the function looks more clearly and beauty.